.eslintrc

```diff
@@ -8,7 +8,7 @@
 	},
 	"rules": {
 		"camelcase": 2,
-		"complexity": [2, 10],
+		"complexity": [2, 11],
```

Comment:
What about creating our own middleware implementation:

```js
app.on('update', function (req, res, next) {
  var requestId = req.requestId;
  var data = req.data;
  var session = res.session;
  res.ok();
  res.send({...});
});  
```

That would provide good separation of concerns:
```js
app.on('update', attachSessionInfo, checkWriteAuthorization, performUpdate);
```

It would make experimenting with new features easier.

At some point I also think it would be good to create expectations:

```js
server.expect('update', 'path/of/update', {expectedData:...});
server.expect(...);
```

And stubs:
```js
server.expect('listen', 'path/blah').sendOk().sendData('path/blah', {data});
```

A good middleware makes it easier to insert that type of feature. (perhaps those ideas code even be a separate project that extends this one).